### DISCUSSION TITLE
Keyboard will show / did show on android incorrect ordering

### DISCUSSION THREAD

**Discussion by iway1** [kirillzyusko/react-native-keyboard-controller#540] [Q&A]
I'm seeing that on android, keyboard did show will be called before keyboard will show, when opening the keyboard. This doesn't make sense, ofc. Seems to happen consistently. Anyone know why this would happen? will show should always be called before did show.

Does this literally just fire an event at the same time as keyboard did show on android?

**Comment by kirillzyusko** [comment]
Hey @iway1 

Would you mind to provide a minimal reproduction example? Which Android version do you use?

Basically `keyboardWillShow` should be called before `keyboardDidShow` (these events even include `timestamp` and `duration` properties). This is how demo-example app works on my Pixel 7 Pro device (Android 14):


https://github.com/user-attachments/assets/992005c7-41cb-4c3f-b6c9-bb8b675fb0be

The popup changes content pretty fast, but this is due to short animation duration on Android (285ms vs 500ms on iOS). Will be happy to answer on your questions 😊 

> The code with demo project can be found at https://github.com/kirillzyusko/react-native-keyboard-controller/blob/main/example/src/screens/Examples/Events/index.tsx

Please also note, that I switch `windowSoftInputMode` to `adjustResize` - you may read [here](https://kirillzyusko.github.io/react-native-keyboard-controller/docs/recipes/platform-differences#android) why it's needed.

**Comment by PandorasBoxTool2024** [comment]
I have also same issue with react native 80.1 and android 15 with sticky keyboard

**Reply by kirillzyusko** [reply]
@PandorasBoxTool2024 could you show the code/video so that I can understand the problem a little bit more?
